src/lib.rs

```diff
@@ -2,28 +2,31 @@
 #![allow(clippy::borrow_deref_ref)]
 
 use std::collections::HashSet;
+use std::num::NonZeroU64;
 use std::thread;
 
 use fancy_regex::Regex;
 use pyo3::exceptions;
 use pyo3::prelude::*;
-use pyo3::types::{PyBytes, PyList, PyTuple};
 use pyo3::PyResult;
+use pyo3::types::{PyBytes, PyList, PyTuple};
 use rustc_hash::FxHashMap as HashMap;
 
+type Rank = u32;
+
 fn _byte_pair_merge<T>(
     piece: &[u8],
-    ranks: &HashMap<Vec<u8>, usize>,
+    ranks: &HashMap<Vec<u8>, Rank>,
```

Comment:
to separate the array indexes from the ranks (and to optimize it slightly, since usize is 64 bits(